make_common/default.mk

```diff
@@ -0,0 +1,32 @@
+# This makefile is for including in projects where you just want to e.g. build
+# all the *.cc files in the current folder.
+#
+# include (path to BoB robotics)/make_common/bob_robotics.mk
+include $(abspath $(dir $(lastword $(MAKEFILE_LIST))))/bob_robotics.mk
+
+SOURCES ?= $(wildcard *.cc)
+OBJECTS := $(foreach obj,$(basename $(SOURCES)),$(OBJECT_PATH)$(obj))
+DEPS    := $(foreach dep,$(basename $(SOURCES)),$(OBJECT_PATH)$(dep).d)
+
+.PHONY: all clean $(EXTRA_DEPS)
+
+all: $(OBJECTS)
+
+-include $(DEPS)
+
+$(OBJECTS): %: %.cc $(DEPS) $(EXTRA_DEPS)
```

Comment:
What does the double : syntax mean?

Reply:
It's needed so you can build the list of programs in ``$(OBJECTS)``: you need the weird syntax I guess because they don't have filename extensions.

And as I'm typing this I realise that variable should be called ``$(EXECUTABLES)`` or something instead!